### ISSUE TITLE
Support WeakRefs

### ISSUE THREAD

**Issue by mdjermanovic** [eslint/eslint#13602]
WeakRefs proposal is now Stage 4.

https://github.com/tc39/proposal-weakrefs

(it seems the proposal's repo hasn't been updated yet with https://github.com/tc39/proposal-weakrefs/pull/212, but it was official in https://github.com/tc39/proposals/commit/489f6dfa9092d84119db8d818a62f23a92f779ba and merged in the spec in https://github.com/tc39/ecma262/commit/71c08971700eeea4f55f49edc6973670c0d4ee2e).

- [X] ESLint Update 1 (minor version)
  - [X] new `es2021` environment with `FinalizationRegistry` and `WeakRef` globals. #13603
- [x] ESLint Update 2 (~major~ minor version):
  - [x] ``no-extend-native`` #14177 **Edit:** per the latest update in our [semver policy](https://github.com/eslint/eslint#semantic-versioning-policy), we can release this change in a minor version.

Not sure if the new syntax label is appropriate since this isn't "syntax".

**Comment by mdjermanovic** [comment]
As for `no-extend-native`, this rule currently doesn't report any errors in the following code:

```js
/* global FinalizationRegistry, WeakRef */

FinalizationRegistry.prototype.foo = bar;
WeakRef.prototype.foo = bar;
```

It will start reporting this code, but not before the next ESLint major version when we will upgrade the `globals` package to `^13` or above.

The same applies to a configuration where the new `es2021` environment (which contains `FinalizationRegistry` and `WeakRef`) is enabled.

So, regardless of user configuration, the `no-extend-native` rule will not disallow modifying `FinalizationRegistry` and `WeakRef` prototypes until ESLint `v8.0.0`.

This is in line with a similar decision in #11803 for new BigInt globals, where it was seen as a breaking change, and thus postponed for `v7.0.0` (though, for most users the change probably happened before `v7.0.0`, because `globals` added BigInt globals in a minor version of the package, and we have `^` dependency).

**Comment by mdjermanovic** [comment]
Marked as breaking for Update 2: upgrading `globals` package, which would cause `no-extend-native` to report more errors.
